### PR TITLE
[1.1.x] led control menu

### PR DESCRIPTION
**Pull request by Tannoo**
Gives control of LEDs from the LCD.

### CODE REVIEW

**Comment by thinkyhead** [comment]
I'm cleaning this up and will push the updated HEAD for your review when it's ready.

**Comment by thinkyhead** [comment]
I've rebased and applied several patches to this PR.
To pull down the latest version (any local changes will be lost):
```sh
git checkout 1.1.x-led_control_menu
git fetch origin
git reset --hard origin/1.1.x-led_control_menu
```
----
Changes include:
- Split up LED-related code similarly to `2.0.x`
- Added a `typedef LEDColor` for use by `set_led_color` and related functions.
- Switched to `LEDColor` for most led-related code.
- Combined strings for better translations. (e.g., in French `blue light` = `lumière bleue`)
- Show **Lights Off** when lights are on, **Lights On** when lights are off.

As with any big set of changes, there may be bugs. Please test whenever you have a chance.

File: Marlin/Configuration_adv.h [hunk 1]
```diff
@@ -482,6 +482,23 @@
...
+//#define LED_CONTROL_MENU
+#if ENABLED(LED_CONTROL_MENU)
+  #define LED_COLOR_PRESETS                 // Enable the Preset Color menu option
+  #if ENABLED(LED_COLOR_PRESETS)
+    #define LED_USER_PRESET_RED       255   // User defined RED value
+    #define LED_USER_PRESET_GREEN     128   // User defined GREEN value
+    #define LED_USER_PRESET_BLUE        0   // User defined BLUE value
+    #define LED_USER_PRESET_WHITE     255   // User defined WHITE value
+    #define LED_USER_PRESET_INTENSITY 255   // User defined intensity
+    //#define LED_USER_PRESET_STARTUP       // Have the printer display the user preset color on startup
```
Comment (thinkyhead) [inline Marlin/Configuration_adv.h]: @Tannoo What is the normal startup state for the LEDs? I assume they are simply turned off by default. Is this option intended to have the LEDs set to a certain color whenever Marlin re-starts (as it does on host connect)? I didn't see any code added to Marlin's `init()` function as a result of this setting, but only an interesting `bool` flag in `ultralcd.cpp`.

**Comment by psavva** [comment]
Maybe an advanced config for setting the start-up led state/colour?

**Comment by thinkyhead** [comment]
> Maybe an advanced config for setting the start-up led state/colour?

A **Set Default LED** menu item would suffice, and then code to save/load the set color from EEPROM.

**Pull request merged**